src/common/helpers/postgres.js

```diff
@@ -0,0 +1,66 @@
+import { Sequelize } from 'sequelize'
+import { Signer } from '@aws-sdk/rds-signer'
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers'
+import { config } from '../../config.js'
+import { healthCheck, createModels } from '../../data/database.js'
+
+export const postgres = {
+  plugin: {
+    name: 'postgres',
+    version: '1.0.0',
+    register: async function (server, options) {
+      server.logger.info('Setting up Postgres')
+
+      if (options.getTokenFromRDS) {
+        options.hooks = {
+          beforeConnect: async (cfg) => {
+            cfg.password = await getToken(options)
+          }
+        }
+      }
+
+      const sequelize = new Sequelize({
+        username: options.user,
+        password: options.passwordForLocalDev,
+        host: options.host,
+        port: options.port,
+        dialect: options.dialect,
+        database: options.database,
+        dialectOptions: {
+          ssl: server.secureContext || false
+        },
+        logging: (msg) => server.logger.info(msg),
+        hooks: options.hooks || {},
+        retry: {
+          backOffBase: 1000,
+          backOffExponent: 1.1,
+          match: [/SequelizeConnectionError/],
+          max: 10,
+          name: 'connection',
+          timeout: 60 * 1000
```

Comment:
We did talk about using a set lifetime of the connection to be lower than the AWS token.  But on looking into it, that's quite difficult todo without risking ending active connections or significantly limiting concurrency.

Have instead reverted to the pattern used on FCP whereby a connection that fails due to a missing token will just retry.  As part of the retry a fresh token is retrieved.